### PR TITLE
Single language indexes + parallel CI build

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
Solving the search index problem (instructions in `build_docs.sh`). 

__Question__:
I would like to ask you guys if you have ideas for a functional substitute to the language selector. Currently we're using the one provided by the `material` theme:
Capture d’écran du 2022-09-19 18-12-50
Problem: it's not going to help us anymore under the following proposal. That's because the selector works just in case there's a list of several languages in the the build config, and under the proposal every build lists but _a single_ language (since a single language per build solves the search index problem). 

So what should we use instead?

### CODE REVIEW

**Comment by m-kuhn** [comment]
I assume we have to inject a list of languages into the build configurations. And create a dropdown from them. Technically, I would have to investigate how to do that.

**Comment by why-not-try-calmer** [comment]
> I assume we have to inject a list of languages into the build configurations. And create a dropdown from them. Technically, I would have to investigate how to do that.

Exactly. One way is through the [plugin](https://github.com/ultrabug/mkdocs-static-i18n#languages) we are already using; we would have to:
1. Determine beforehand the list of languages we want to build for (margin for optimization: we could determine this after pull the languages with transifex to apply the convention of not using any language below the 30% translation threshold).
2. Generate, for every language, a `mkconfig.yml` file with the following:
```
    - i18n:
      default_language: en
      languages:
        CURRENT_LANG_CODE:
          name: CURRENT_LANG_NAME
          build: true
          link: CURENT_LANG_CODE
        :
          name: ANY_OTHER_LANG_NAME
          build: false
          link: ANY_OTHER_LANG_LANG_CODE
```
At the end of the day, we would still get one different build for each language, but the builds would be accessible to one another via the selector generated from these settings. Not tested though, it's just an idea.

The other would of doing this woud be to create in-house the selector from the internal links to the language. Using the selector would thus allow the user to move from one build to another.

**Comment by m-kuhn** [comment]
So the "complexity" is injecting the language list into yaml files? Sounds good to me 👍
We can start with a hardcoded list

**Comment by why-not-try-calmer** [comment]
> complex



> So the "complexity" is injecting the language list into yaml files? Sounds good to me +1 We can start with a hardcoded list

The complexity I was wondering about was the complexity of the second approach: customizing the theme to write from scratch a functional language selector to work (needs to be custom because it must be aware of the 1 build = 1 language file structure). 

If one wonders about the complexity of the first approach, I think it's reasonable, and easier to get right from the outset, so I'll try it first.

File: mkdocs.yml [hunk 1]
```diff
@@ -156,26 +155,26 @@ extra:
       link: mailto:info@opengis.ch
   alternate:
     - name: English
-      link: /en/
+      link: /
       lang: en
     - name: Deutsch
-      link: /de/
+      link: /de/de/
```
Comment (m-kuhn) [inline mkdocs.yml:161]: Can we avoid this doubling?

Comment (m-kuhn) [reply mkdocs.yml:161]: Thanks for clarifying 👍

**Comment by why-not-try-calmer** [comment]
I seem to have fixed it with the latest commit but translations are not properly applied when building. Also the git history suggests that translation strings have been missing _en masse_ for a while. I've got in touch with Mario; I hope I can fix this issue because it might be spoiling this PR.

**Comment by why-not-try-calmer** [comment]
Put on hold as we hope to add only well-translated languages so that overlaps are too rare. Will reopen if needed.

**Pull request closed without merging**